website/src/components/Stats/Stats.components.tsx

```diff
@@ -38,30 +39,191 @@ type MessageTreeStateStatsProps = {
   };
 };
 
+function getTableData(
+  stat: StatType & {
+    stats: {
+      lang: string;
+      state: string;
+      count: number;
+    }[];
+  },
+  t
+) {
+  // The headers come from the message tree State.
+  // These two states are not considered because they make
+  // the table too wide and they are not so interesting:
+  // - "ready_for_scoring",
+  // - "scoring_failed"
+  let headers = [
+    "language", // This one doesn't come from the message tree state
+    "prompt_lottery_waiting",
+    "initial_prompt_review",
+    "growing",
+    "backlog_ranking",
+    "ranking",
+    "ready_for_export",
+    "aborted_low_grade",
+    "halted_by_moderator",
+  ];
+
+  const rows = [];
+
+  for (let i = 0; i < stat.stats.length; i++) {
+    const lang = stat.stats[i].lang;
+
+    let langRow = rows.findIndex((row) => row[0] === lang);
+
+    if (langRow === -1) {
+      // The language is not yet in the data. Add it
+      rows.push(Array(headers.length).fill(0));
+      langRow = rows.length - 1;
+      rows[langRow][0] = lang;
+    }
+
+    const state = stat.stats[i].state;
+    const stateCol = headers.indexOf(state);
+    if (stateCol === -1) {
+      continue;
+    }
+
+    const threadCount = stat.stats[i].count;
+    rows[langRow][stateCol] = threadCount;
+  }
+
+  headers = headers.map((item) => t(getTypeSafei18nKey(item)));
+
+  // Add the language code in brackets after the language name
+  rows.forEach((row) => {
+    const langCode = row[0];
+    row[0] = getLocaleDisplayName(langCode) + " [" + langCode + "]";
```

Comment:
You could use a template literal here for clarity.

Reply:
Done.